src/pip/_internal/vcs/git.py

```diff
@@ -370,20 +370,25 @@ def update_submodules(cls, location):
         )
 
     @classmethod
-    def controls_location(cls, location):
-        if super(Git, cls).controls_location(location):
-            return True
+    def get_repository_root(cls, location):
+        loc = super(Git, cls).get_repository_root(location)
+        if loc:
+            return loc
         try:
-            r = cls.run_command(['rev-parse'],
-                                cwd=location,
-                                show_stdout=False,
-                                on_returncode='ignore',
-                                log_failed_cmd=False)
-            return not r
+            r = cls.run_command(
+                ['rev-parse', '--show-toplevel'],
+                cwd=location,
+                show_stdout=False,
+                on_returncode='raise',
+                log_failed_cmd=False,
+            )
         except BadCommand:
             logger.debug("could not determine if %s is under git control "
                          "because git is not available", location)
-            return False
+            return None
+        except InstallationError:
```

Comment:
This seems weird, I'd say `call_subprocess` should not raise `InstallationError` which is too high-level.

Reply:
It is definitely not right, but I’d say this belongs to another (refactoring) PR. Note that the Mercurial implementation already catches `InstallationError` right now.